colossalai/initialize.py

```diff
@@ -267,10 +270,27 @@ def initialize(model: nn.Module,
     if verbose:
         logger.info(f"cuDNN benchmark = {cudnn_benchmark}, deterministic = {cudnn_deterministic}", ranks=[0])
 
-    # first sync model across dp ranks
-    model.to(get_current_device())
-    use_zero3 = hasattr(gpc.config, 'zero') and gpc.config.zero.level == 3
-    if not moe_env.is_initialized() and not use_zero3:
+    use_zero = hasattr(gpc.config, 'zero')
+    if use_zero:
+        zero_cfg = gpc.config.get('zero', None)
+        if zero_cfg is not None:
+            cfg_ = zero_cfg.copy()
+        else:
+            cfg_ = {}
+        model, optimizer = convert_to_zero_v2(model_builder=model, optimizer_class=torch.optim.Adam, **cfg_)
```

Comment:
Why fix `optimizer_class` to `torch.optim.Adam`?

Reply:
pass it via config